mne/io/brainvision/brainvision.py

```diff
@@ -136,12 +138,38 @@ def __init__(self, vhdr_fname, montage=None,
                 offsets = None
                 n_samples = n_samples // (dtype_bytes * (info['nchan'] - 1))
         self.preload = False  # so the event-setting works
+
+        # Get annotations from vmrk file and then use events_from_annotations
+        trig_shift_by_type = _check_trig_shift_by_type(trig_shift_by_type)
+        annots = read_annotations_brainvision(mrk_fname, info['sfreq'])
+        event_id = dict() if event_id is None else event_id
+        event_id = partial(_event_id_func, event_id=event_id,
+                           trig_shift_by_type=trig_shift_by_type)
+
+        # XXX : We use a mock raw as self is not proper yet
+        # and it's not possible to do it after as one needs the events
+        # to create the stim channel before calling super init
```

Comment:
I understand why we are doing this. But this seems weird.

Reply:
Maybe there should be a lower-level `_events_from_annotations` that takes annotations (not raw), sfreq and optionally first_samp? Or would it not be enough?

Reply:
Also perhaps add a note that once we no longer synthesize a STIM channel at all (this is planned to be deprecated, right?) this can therefore be refactored to be simpler and/or removed.

Incidentally another way to work around this would be to:

1. create the stim channel as a dummy instead, i.e. `np.zeros`
2. do `BaseRaw.__init__`
3. `set_annotations`
4. `events_from_annotations`
5. modify/correct the stim channel values as necessary

Reply:
yes we'll no longer need this once we stock synthesizing the stim channel 

@mmagnuski I considered the use of a private _events_from_annotations too but makes more complex the raw object (change how raw.time_as_index is implemented by the use of private auxiliary functions) for a fix that is temporary.

@larsoner what you suggest requires computation . My hack with mock object does not generate any new computational cost.

Reply:
I don't see any additional computation in my solution that isn't already being done here, can you be more specific?

Reply:
The modify/correct step is really more "replace the dummy values", so this is a trivial memcpy sort of operation